lifelines/utils/__init__.py

```diff
@@ -275,11 +287,49 @@ def restricted_mean_survival_time(model_or_survival_function, t: float = np.inf)
             sf = sf.reset_index()
             return (sf["index"].diff().shift(-1) * sf[model._label]).sum()
         else:
-            return quad(model.survival_function_at_times, 0, t)[0]
+            return quad(model.predict, 0, t)[0]
     else:
         raise ValueError("Can't compute RMST of object %s" % model_or_survival_function)
 
 
+def _expected_value_of_survival_squared_up_to_t(model_or_survival_function, t: float = np.inf) -> float:
+    r"""
+    Compute the restricted mean survival time, RMST, of a survival function. This is defined as
+
+    .. math::  E[t]^2 = 2 \int_0^t \tau S(\tau) d\tau
+
+    For reason why we use an upper bound and not always :math:`\infty` is because the tail of a survival
+    function has high variance and strongly effects the RMST.
+
+    Parameters
+    -----------
+
+    model_or_survival_function: lifelines model or DataFrame
+        This can be a univariate model, or a pandas DataFrame. The former will provide a more accurate estimate however.
+    t: float
+        The upper limit of the integration in the RMST.
+
+
+    References
+    -------
+    https://bmcmedresmethodol.biomedcentral.com/articles/10.1186/1471-2288-13-152#Sec27
+
+    """
+    import lifelines
+
+    if isinstance(model_or_survival_function, pd.DataFrame):
+        sf = model_or_survival_function.loc[:t]
+        sf = sf.append(pd.DataFrame([1], index=[0], columns=sf.columns)).sort_index()
+        sf_tau = sf * sf.index.values[:, None]
+        return 2 * trapz(y=sf_tau.values[:, 0], x=sf_tau.index)
+    elif isinstance(model_or_survival_function, lifelines.fitters.UnivariateFitter):
+        # lifelines model
+        model = model_or_survival_function
+        return 2 * quad(lambda tau: (tau * model.predict(tau)), 0, t)[0]
```

Comment:
hm, this 0 is going to give the wrong numbers for obs that have negative duration (not common but plausible)